### DISCUSSION TITLE
Allow TeslaMate to temporarily switch off the access to Tesla API

### DISCUSSION THREAD

**Discussion by petariga** [teslamate-org/teslamate#791] [Ideas]
**What would you like to be added**:
Just **on/off switch** for temporarily disabling the access to Tesla API would be nice.

**Why is this needed**:
My car has spent quite a lot of time at the Tesla service center due to a software glitch in the recent months. Each time the car was handed over to technicians for monitoring they have always complained about 3rd party software accessing the Tesla API and therefore causing to keep the car alive/on-line. It was impossible to explain them that the TeslaMate has no connection to the glitch.

I needed to get to console to access TeslaMate running environment and shut it down. I wasn’t able to show them the graphs from TeslaMate afterwards, because Grafana was shut down as well. So it was inconvenient. If there was a switch to enable/disable API access from within the TeslaMate setting page, it would be much easier to control the access to Tesla API and it would allow me to continue to display the graphs through Grafana even when the access has to be switched off.

**Comment by jmiverson** [comment]
Seems like a reasonable ask, but also sort of niche. If nothing happens, and you’re running the Docker install method, you can always stop just the Teslamate container and leave Grafana and the Database containers running. Then you can access all the reports and stuff without the Teslamate container connecting to the Tesla API.

**Comment by petariga** [comment]
So far, if I stop the Docker with suggested “docker-compose down”, it stops everything. I also need a console/ssh access to do that. I am not telling it is not doable, it is just inconvenient. Having the option directly in the TeslaMate settings would work much better, is easily reachable through a web browser and as far as I understand, it should not be as complicated for implementation.

**Comment by jmiverson** [comment]
Instead of down, use “stop” and include the name of the container afterward. 

`docker-compose stop teslamate`

**Comment by petariga** [comment]
Thank you, this works to split the shutdown between TeslaMate/Grafana, however, the access to console/ssh is still required.

**Comment by bahree** [comment]
I could see a toggle in the setting useful - and in other reasons - e.g. parked long term when travelling (airport for example) - of course when things go back to normal(ish).

**Comment by JakobLichterfeld** [comment]
> I could see a toggle in the setting useful - and in other reasons - e.g. parked long term when travelling (airport for example) - of course when things go back to normal(ish).

I could not see a reason why I would stop logging when parked. Teslamate does not wake the vehicle only, so no additional consumption. Benefits like vampire drain and so on if logging all time

**Comment by petariga** [comment]
@JakobLichterfeld unfortunately, this is true only in the ideal case when everything works perfectly. If something fails and you have to debug, you realize that TeslaMate for example extends the on-line period of the monitored car from 4 minutes to approx. 17 minutes, therefore causing higher than necessary vampire drain too. I can prove it from my experience and it has been confirmed by adriankumpf, see the comments in the issue [571](https://github.com/adriankumpf/teslamate/issues/571).

**Comment by JakobLichterfeld** [comment]
>  If something fails and you have to debug, you realize that TeslaMate for example extends the on-line period of the monitored car from 4 minutes to approx. 17 minutes, therefore causing higher than necessary vampire drain too. 

For sure, in older realeases you can change settings for time before sleeping. In newer there is no such a setting when using streaming api. So car wakes up every 2 days or so to recharge 12V battery, this is not teslamate it is Tesla powernet design. Teslamate can extend the wake period a little, no real influence on milage

**Comment by petariga** [comment]
@JakobLichterfeld you probably did not read the issue #571 . The car can behave in a way that it repeatedly wakes up and therefore the vampire drain from extending the on-line period from 4 minutes to 17 minutes can be very significant (about 2-3x). As I wrote, what you state is only true, if everything works perfectly. Once you face any issues, the situation with vampire drain can change a lot.

**Comment by JakobLichterfeld** [comment]
I read it, so extension by 3 minutes. When not a WiFi or key nearby nor a dead 12 V battery I do not see the need of such a switch. If someone really want to stop teslamate you can do it with `docker-compose stop teslamate`

**Comment by petariga** [comment]
It wasn't the extension of the on-line state by 3 minutes, but from 4 minutes to 17 minutes, which had caused 2-3 times higher drain. 

Why do I ask for a toggle switch? Because it would be convenient for some people in some scenarios. I understand that much more things can be always done just with the console commands. Why do we use TeslaMate? Because we can learn more about the car and it makes our life with it easier and more convenient. If I didn't get the philosophy behind TeslaMate right, you can always leave this idea behind.

Adding this API access on/off switch to TeslaMate settings would take the programmer shorter time than our discussion here ;-). Moreover, it will not harm people who do not want to use it :-).

**Comment by JakobLichterfeld** [comment]
> Adding this API access on/off switch to TeslaMate settings would take the programmer shorter time than our discussion here ;-).

So Pull Request are always welcome by @adriankumpf  ;)

**Comment by JakobLichterfeld** [comment]
> Moreover, it will not harm people who do not want to use it :-).

I do predict some open issues why is teslamate not logging ;)

**Comment by petariga** [comment]
My days of programming are long over and I do not want to put a mess into Adrian's code :-)
